modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -496,19 +495,15 @@ public void start() {
 
             lowWatermark.start();
 
-            startTables();
+            CompletableFuture<Long> recoveryFinishFuture = metaStorageMgr.recoveryFinishedFuture();
 
-            try {
-                metaStorageMgr.recoveryFinishedFuture()
-                        .thenComposeAsync(this::performRebalanceOnRecovery, ioExecutor)
-                        .get();
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-
-                throw new IgniteInternalException(INTERNAL_ERR, e);
-            } catch (ExecutionException e) {
-                throw new IgniteInternalException(INTERNAL_ERR, e);
-            }
+            assert recoveryFinishFuture.isDone();
+
+            long recoveryRevision = recoveryFinishFuture.join();
+
+            startTables(recoveryRevision);
+
+            performRebalanceOnRecovery(recoveryRevision);
```

Comment:
Can one of these methods return futures and add to `startVv.update(recoveryRevision, (v, e) -> allOf(startTablesFuture, performRebalanceOnRecoveryFuture));` once?

Reply:
They can, but is it really better? The current approach is also fine, I think